Refactor more speech functions

feerrenrut opened the pull request:
<!--
Please fill in the following template, for an explanation of the sections see:
https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples
-->

### Link to issue number:
None

### Summary of the issue:
Returning sequences from these functions allows us greater flexibility for affecting all speech that is caused by a call to speakObject.

### Description of how this pull request fixes the issue:
Split several functions into a speakX and getXspeech

This PR maintains backwards compatibility

### Testing performed:
Ran NVDA from source.

### Known issues with pull request:
- Todo: rename getSpeechForSpelling to getSpellingSpeech

### Change log entry:

Section: New features, Changes, Bug fixes

feerrenrut committed:
Fix getTextWithFields docstring

feerrenrut committed:
Encapsulate speakWithoutPauses in a class

For easier testing / refactoring.

feerrenrut committed:
Rename _speakTextInfo_addMath -> _extendSpeechSequence_addMathForTextInfo

Clarify that the sequence is modified in place, nothing is spoken.

feerrenrut committed:
Split speakTextInfo: getTextInfoSpeech is a generator with return value.

- Speech can happen as it is generated, with the same number of calls
  to speak as previous implementation.
- Returns the same boolean values as previous implementation.
- To capture these return values, use GeneratorWithReturn class.

feerrenrut committed:
Add return type annotation: getSpeechForSpelling

feerrenrut committed:
Split speakObjectProperties to return a speech sequence

feerrenrut committed:
Split speakMessage to return speechSequence

feerrenrut committed:
Split speakSelectionMessage and speakPreselectedText to return speech sequences

feerrenrut committed:
Extract 'allowed properties' calc from speakObject

Simplify the speakObject function by extracting this calculation into a
helper.

feerrenrut committed:
Format extracted 'allowed properties' calc

The new method '_objectSpeech_calculateAllowedProps()',
was extracted from 'speakObject()' and needs formatting
to pass linting checks.

feerrenrut committed:
Split speakObject to return a speech sequence

Must flatten the results from getTextInfoSpeech generator.
Rename _speakPlaceholderIfEmpty -> _getPlaceholderSpeechIfTextEmpty

zstanecic commented:
Hi Reef,

Will this change affect drivers, which are already ported to py3?

Note that codefactory closed porting of their voalizer.

Will this affect RHVoice or any other tts?

We should start thinking about reputation among users.

The more backward compatible changes we have, the less is the possibility that they will want to upgrade to the new NVDA version.

Now we have this situation with py3.

Isn’t it better to leave this for NVDA 2020?

This is just my thinking and questions, along with concerns.

Sorry, but if we start to bring more and more backward compatible changes now, it will break us, and we will be in the situation that will lead to fragmentation, and people will prefer staying on NVDA 2019.2.1, because it’s older and add-ons work flawlessly.

 

From: Reef Turner <notifications@github.com> 
Sent: Sunday, December 8, 2019 4:18 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [nvaccess/nvda] Refactor more speech functions (#10593)

 


Link to issue number:


None


Summary of the issue:


Returning sequences from these functions allows us greater flexibility for affecting all speech that is caused by a call to speakObject.


Description of how this pull request fixes the issue:


Split several functions into a speakX and getXspeech


Testing performed:


Ran NVDA from source.


Known issues with pull request:


*	Todo: rename getSpeechForSpelling to getSpellingSpeech


Change log entry:


Section: New features, Changes, Bug fixes

  _____  


You can view, comment on, or merge this pull request online at:


  https://github.com/nvaccess/nvda/pull/10593


Commit Summary


*	Fix getTextWithFields docstring
*	Split speaking and getSpeechWithoutPauses
*	rename _speakTextInfo_addMath -> _extendSpeechSequence_addMathForTextInfo
*	Split speakTextInfo to return speech sequence
*	Add return type to getSpeechForSpelling
*	Split speakObjectProperties to return a speech sequence
*	Split speakMessage to return speechSequence
*	Split speakSelectionMessage and speakPreselectedText to return speech sequences
*	Split out 'allowed properties' calculation from speakObject
*	Format _objectSpeech_calculateAllowedProps
*	Split speakObject to return a speech sequence
*	Fix generator annotation
*	Fix lint


File Changes


*	M source/speech/__init__.py <https://github.com/nvaccess/nvda/pull/10593/files#diff-0>  (481) 
*	M source/textInfos/__init__.py <https://github.com/nvaccess/nvda/pull/10593/files#diff-1>  (8) 


Patch Links:


*	https://github.com/nvaccess/nvda/pull/10593.patch
*	https://github.com/nvaccess/nvda/pull/10593.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10593?email_source=notifications&email_token=ACVCDE4BPSUA4S6BHSM5G2LQXUF2RA5CNFSM4JX4652KYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4H64VYCA> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACVCDE6G3VXC6MDZNCYRTU3QXUF2RANCNFSM4JX4652A> .  <https://github.com/notifications/beacon/ACVCDE7G545NOYOTAKQBQVLQXUF2RA5CNFSM4JX4652KYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4H64VYCA.gif>

michaelDCurran commented:
@zstanecic : These changes should not affect synthDrivers at all.

Brian1Gaff commented:
There should, in my view be a neutral view on this, at least for some time, 
since there will be more annoyance from people who have updated whose 
existing maybe home scripted solutions suddenly stop working than from the 
fragmentation of users. After all, I and no doubt others still have some old 
hardware running XP as useful extras, which use a very old version of nvda, 
and that has not really worried the progression much, as long as people keep 
their old add ons.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message ----- 
From: "zstanecic" <notifications@github.com>
To: "nvaccess/nvda" <nvda@noreply.github.com>
Cc: "Subscribed" <subscribed@noreply.github.com>
Sent: Sunday, December 08, 2019 4:12 PM
Subject: Re: [nvaccess/nvda] Refactor more speech functions (#10593)


Hi Reef,

Will this change affect drivers, which are already ported to py3?

Note that codefactory closed porting of their voalizer.

Will this affect RHVoice or any other tts?

We should start thinking about reputation among users.

The more backward compatible changes we have, the less is the possibility 
that they will want to upgrade to the new NVDA version.

Now we have this situation with py3.

Isn’t it better to leave this for NVDA 2020?

This is just my thinking and questions, along with concerns.

Sorry, but if we start to bring more and more backward compatible changes 
now, it will break us, and we will be in the situation that will lead to 
fragmentation, and people will prefer staying on NVDA 2019.2.1, because it’s 
older and add-ons work flawlessly.



From: Reef Turner <notifications@github.com>
Sent: Sunday, December 8, 2019 4:18 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [nvaccess/nvda] Refactor more speech functions (#10593)




Link to issue number:


None


Summary of the issue:


Returning sequences from these functions allows us greater flexibility for 
affecting all speech that is caused by a call to speakObject.


Description of how this pull request fixes the issue:


Split several functions into a speakX and getXspeech


Testing performed:


Ran NVDA from source.


Known issues with pull request:


* Todo: rename getSpeechForSpelling to getSpellingSpeech


Change log entry:


Section: New features, Changes, Bug fixes

  _____


You can view, comment on, or merge this pull request online at:


  https://github.com/nvaccess/nvda/pull/10593


Commit Summary


* Fix getTextWithFields docstring
* Split speaking and getSpeechWithoutPauses
* rename _speakTextInfo_addMath -> _extendSpeechSequence_addMathForTextInfo
* Split speakTextInfo to return speech sequence
* Add return type to getSpeechForSpelling
* Split speakObjectProperties to return a speech sequence
* Split speakMessage to return speechSequence
* Split speakSelectionMessage and speakPreselectedText to return speech 
sequences
* Split out 'allowed properties' calculation from speakObject
* Format _objectSpeech_calculateAllowedProps
* Split speakObject to return a speech sequence
* Fix generator annotation
* Fix lint


File Changes


* M source/speech/__init__.py 
<https://github.com/nvaccess/nvda/pull/10593/files#diff-0>  (481)
* M source/textInfos/__init__.py 
<https://github.com/nvaccess/nvda/pull/10593/files#diff-1>  (8)


Patch Links:


* https://github.com/nvaccess/nvda/pull/10593.patch
* https://github.com/nvaccess/nvda/pull/10593.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub 
<https://github.com/nvaccess/nvda/pull/10593?email_source=notifications&email_token=ACVCDE4BPSUA4S6BHSM5G2LQXUF2RA5CNFSM4JX4652KYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4H64VYCA> 
, or unsubscribe 
<https://github.com/notifications/unsubscribe-auth/ACVCDE6G3VXC6MDZNCYRTU3QXUF2RANCNFSM4JX4652A> 
. 
<https://github.com/notifications/beacon/ACVCDE7G545NOYOTAKQBQVLQXUF2RA5CNFSM4JX4652KYY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4H64VYCA.gif>



-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/nvaccess/nvda/pull/10593#issuecomment-562964673

feerrenrut commented:
@zstanecic and @Brian1Gaff This PR does not break backwards compatibility. The description has been updated to make that clear.

leonardder commented on source/speech/__init__.py:
```diff
@@ -379,34 +412,44 @@ def speakObjectProperties(  # noqa: C901
+		reason,
+		_prefixSpeechCommand,
+	)
+	speak(sequence, priority=priority)
```
May be do a similar if check as in the other functions?

LeonarddeR reviewed (approved):
This looks good to me, however note that I only had to had time to look at the code briefly, I haven't tested this. I therefore recommend a look from @michaelDCurran before merging this.

michaelDCurran reviewed (changes requested):
* Would it be possible to make a new  type annotation for 'reason'? All through this code you have reason: str. It would be great if this could be its own type. 
* with all the speakSelection functions:  it would be good if they could take speech sequences, rather than just strings.  Then speakSelectionChange can pass in a sequence from getTextInfoSpeech rather than just TextInfo.text. It also means that some of the code in speakSelectionChange can be replaced with a call to getSpellingSpeech. However, these changes would mean we can't have the text included with the message as part of one translatable string. Though I don't believe this is a bad thing.

feerrenrut committed:
Add debug logging and formatting to sayAllHandler

feerrenrut committed:
Rename getSpeechForSpelling -> getSpellingSpeech

For now deprecate (but maintain) the old name.

feerrenrut committed:
Use flush instead of directly manipulating member var

feerrenrut committed:
Add names for callback commands to give context in logging.

feerrenrut committed:
Extract stages of speakWithoutPauses

feerrenrut committed:
Remove recursive calls

Use instance methods directly.

feerrenrut committed:
Refactor getSpeechWithoutPauses to use generators

Make speakWithoutPauses an instance method
- To make it more testable.
- Keep backwards compat; speakWithoutPauses name is added at module level.

Make getSpeechWithoutPauses a generator with return value.
- Speech can happen as it is generated, with the same number of calls
  to speak as previous implementation.
- Returns the same boolean values as previous implementation.
- To capture these return values, use GeneratorWithReturn class.

Add unit test for speechWithoutPauses
 - To ensure behavior has not changed between implementations

michaelDCurran commented:
Is it just me, or have the changes to ```speech/__init__.py``` been lost in this pr?

feerrenrut commented:
> Is it just me, or have the changes to speech/__init__.py been lost in this pr?

I think it's just you 😉. It is hidden from the diff initially because it is "large". You can jump to it, or other files by pressing 't' and typing '__init__'. A direct link to the file in the diff: https://github.com/nvaccess/nvda/pull/10593/files#diff-46301b9ce4a78abc208b3358db03e972

feerrenrut commented:
I'm going to rebase and merge this PR (to preserve the history of the refactor). I'll push after the rebase so that the commits in this PR / branch match.
